REGRESSION (274503@main): ynet.co.il: Video in element fullscreen is cropped

aestes opened the pull request:
#### 82a00cb3625eebcfc66e15c0355cf565a8de158f
<pre>
REGRESSION (274503@main): ynet.co.il: Video in element fullscreen is cropped
<a href="https://bugs.webkit.org/show_bug.cgi?id=295219">https://bugs.webkit.org/show_bug.cgi?id=295219</a>
<a href="https://rdar.apple.com/150685881">rdar://150685881</a>

Reviewed by Wenson Hsieh.

In 274503@main we enforced a min/max/current zoom scale value of 1 on WKScrollView when in element
fullscreen. This ensures that if the user had pinched to zoom in on a page, that scale factor was
undone when entering fullscreen. However, 1 is not the correct scale value to apply for a page that
is wider than the screen&apos;s dimensions, and on such pages web content would appear cropped in element
fullscreen.

Addressed this by setting the min/max/current zoom scale to the initial zoom scale in element
fullscreen. This ensures that a wide page is scaled to fit the screen.

Added an API test.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _updateScrollViewForTransaction:]):
(-[WKWebView _forcesInitialScaleFactor]):
(-[WKWebView _setForcesInitialScaleFactor:]):
(-[WKWebView _overrideZoomScaleParametersWithMinimumZoomScale:maximumZoomScale:allowUserScaling:]): Deleted.
(-[WKWebView _clearOverrideZoomScaleParameters]): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(WebKit::WKWebViewState::applyTo):
(-[WKFullScreenWindowController _enterFullScreen:windowScene:completionHandler:]):
(-[WKFullScreenWindowController _reinsertWebViewUnderPlaceholder]):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UI/ElementFullscreen.mm: Added.
(-[TestElementFullscreenDelegate waitForDidEnterElementFullscreen]):
(-[TestElementFullscreenDelegate _webViewDidEnterElementFullscreen:]):
(TEST(ElementFullscreen, ScrollViewSetToInitialScale)):

Canonical link: <a href="https://commits.webkit.org/296848@main">https://commits.webkit.org/296848@main</a>
</pre>

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5a0deee3ff5a970905104bbbac6dc10c8c3dea2d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47388

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c9384db6a43acca58249860bd64567ab77791da2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47388

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/75cced64513d4f4b6f45ef5e70ecb04107cabfe6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47388

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4b29884047b73c3cb087f8a3ee46ffb16928d912. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47388

aestes committed:
REGRESSION (274503@main): ynet.co.il: Video in element fullscreen is cropped
https://bugs.webkit.org/show_bug.cgi?id=295219
rdar://150685881

Reviewed by Wenson Hsieh.

In 274503@main we enforced a min/max/current zoom scale value of 1 on WKScrollView when in element
fullscreen. This ensures that if the user had pinched to zoom in on a page, that scale factor was
undone when entering fullscreen. However, 1 is not the correct scale value to apply for a page that
is wider than the screen's dimensions, and on such pages web content would appear cropped in element
fullscreen.

Addressed this by setting the min/max/current zoom scale to the initial zoom scale in element
fullscreen. This ensures that a wide page is scaled to fit the screen.

Added an API test.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _updateScrollViewForTransaction:]):
(-[WKWebView _forcesInitialScaleFactor]):
(-[WKWebView _setForcesInitialScaleFactor:]):
(-[WKWebView _overrideZoomScaleParametersWithMinimumZoomScale:maximumZoomScale:allowUserScaling:]): Deleted.
(-[WKWebView _clearOverrideZoomScaleParameters]): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(WebKit::WKWebViewState::applyTo):
(-[WKFullScreenWindowController _enterFullScreen:windowScene:completionHandler:]):
(-[WKFullScreenWindowController _reinsertWebViewUnderPlaceholder]):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UI/ElementFullscreen.mm: Added.
(-[TestElementFullscreenDelegate waitForDidEnterElementFullscreen]):
(-[TestElementFullscreenDelegate _webViewDidEnterElementFullscreen:]):
(TEST(ElementFullscreen, ScrollViewSetToInitialScale)):

Canonical link: https://commits.webkit.org/296848@main

webkit-commit-queue commented:
Committed 296848@main (82a00cb3625e): <https://commits.webkit.org/296848@main>

Reviewed commits have been landed. Closing PR #47388 and removing active labels.
